config/v1/types_infrastructure.go

```diff
@@ -505,9 +505,113 @@ type BareMetalPlatformStatus struct {
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
 }
 
+type OpenStackAPIBGPPeer struct {
+	// asn is the ASN of the peer.
+	//
+	// +kubebuilder:validation:Required
+	ASN string `json:"asn"`
+
+	// ip is the IP address of the peer. It may be either IPv4 or IPv6.
+	//
+	// +kubebuilder:validation:Required
+	IP string `json:"ip"`
+}
+
+// OpenStackAPIBGPSpeaker species
+type OpenStackAPIBGPSpeaker struct {
+	// failureDomain is the name of a failure domain which this BGP
+	// configuration applies to.
+	//
+	// +kubebuilder:validation:Required
+	FailureDomain string `json:"failureDomain"`
+
+	// asn specifies the ASN to be used by the BGP speaker.
+	//
+	// +kubebuilder:validation:Required
+	ASN string `json:"asn"`
+
+	// peers is a list of all BGP peers of this speaker.
+	//
+	// +kubebuilder:validation:MinItems:=1
+	Peers []OpenStackAPIBGPPeer `json:"peers"`
+}
+
+type OpenStackAPIBGPConfiguration struct {
+	// apiVIPs is a list of IP addresses which will be published via BGP for
+	// use by the API server. It may contain both IPv4 and IPv6 addresses.
+	//
+	// +kubebuilder:validation:MinItems:=1
+	APIVIPs []string `json:"apiVIPs"`
+
+	// speakers is a list of BGP speaker configurations. We require a
+	// speaker configuration for every failure domain hosting a control
+	// plane node.
+	//
+	// +kubebuilder:validation:MinItems:=1
+	// +listType=map
+	// +listMapKey=failureDomain
+	Speakers []OpenStackAPIBGPSpeaker `json:"speakers"`
+}
+
+// OpenStackFailureDomain specifies a failure domain for an OpenStack server.
+// Specifically, it specifies a set of values which will be set on all machines
+// using the failure domain.
+type OpenStackFailureDomain struct {
+	// name is the unique name of this failure domain.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// computeZone specifies the OpenStack Compute availability zone for
+	// all servers in this failure domain.
+	//
+	// If not specified, the servers are provisioned in the default Compute
```

Comment:
Is this an OpenStack default?

Reply:
Yes. It's not something we can substitute.

@pierreprinetti I think we should clean this language up though. It implies there's a default AZ which all servers will be put into, which isn't the case. Perhaps:

```
If not specified the servers are provisioned without reference to availability zones. Each server will be provisioned in any availability zone with sufficient capacity.
```

Reply:
> Each server will be provisioned in any availability zone with sufficient capacity

Isn't this something you can configure as the OpenStack admin (there's a reference to a default "nova" zone [in the docs](https://docs.openstack.org/nova/latest/admin/availability-zones.html))?

However, I understand that "the default Compute zone" is misleading.

What about:

```
If not specified the servers are provisioned without reference to availability zones. Server placement is delegated to the OpenStack defaults.
```

Reply:
No. 'nova' is just the name of a default AZ for any compute which isn't explicitly in some other AZ.

However, I prefer your version. It's more concise.

Reply:
addressed